cmd/waku/server/rest/relay.go

```diff
@@ -171,8 +171,13 @@ func (r *RelayService) getV1Messages(w http.ResponseWriter, req *http.Request) {
 		w.WriteHeader(http.StatusBadRequest)
 		return
 	}
-
-	var err error
+	topic, err := url.QueryUnescape(topic)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		_, err = w.Write([]byte("invalid topic format"))
+		r.log.Error("writing response", zap.Error(err))
+		return
+	}
```

Comment:
@chaitanyaprem, @harsh-98,  what do you think if we move the code from https://github.com/waku-org/go-waku/pull/856/files#diff-66b9a518fabf46f0f294417e83bdd3e2305e7db3f7bc3a82a0425fa142968617R367-R381 into rest/utils.go so both filter and relay can use it?

Reply:
Makes sense, did not think this code was already there. 
Will do the change.

Reply:
That PR is approved, we can go ahead merge it if it looks okay to you, and then rebase this one to contain that change! In the meantime i'll approve this, since this fix solves the problem anyway